src/swap/saga.ts

```diff
@@ -0,0 +1,109 @@
+import { CeloTx } from '@celo/connect'
+import { TxParamsNormalizer } from '@celo/connect/lib/utils/tx-params-normalizer'
+import { ContractKit } from '@celo/contractkit'
+import { call, put, takeEvery } from 'redux-saga/effects'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import {
+  swapApprove,
+  swapError,
+  swapExecute,
+  swapPriceChange,
+  swapStart,
+  swapSuccess,
+} from 'src/swap/slice'
+import { sendTransaction } from 'src/transactions/send'
+import { newTransactionContext } from 'src/transactions/types'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import Logger from 'src/utils/Logger'
+import { getContractKit } from 'src/web3/contracts'
+import networkConfig from 'src/web3/networkConfig'
+import { getConnectedUnlockedAccount } from 'src/web3/saga'
+import { applyChainIdWorkaround, buildTxo } from 'src/web3/utils'
+
+const TAG = 'swap/saga'
+
+function getPriceDiff(price1: number, price2: number) {
+  return (Math.abs(price1 - price2) / ((price1 + price2) / 2)) * 100
+}
+
+export function* swapSubmitSaga(data: any) {
+  try {
+    // Navigate to swap pending screen
+    yield call(navigate, Screens.SwapPending)
+
+    // Check that prices are within 2% of each other
+    const priceDiff: number = yield call(
+      getPriceDiff,
+      +data.payload.unvalidatedSwapTransaction.price,
+      +data.payload.unvalidatedSwapTransaction.guaranteedPrice
+    )
+    if (priceDiff >= 2) {
```

Comment:
oh i understand the 2 is the slippage tolerance? can this be a config variable?

Reply:
Fixed in debfbea4e50ffa7eb1b7ecc0be6b08e84fba5bea!